Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -162,9 +162,17 @@ class BrowserViewController: UIViewController {
         return keyboardPressesHandlerValue
     }
 
-    fileprivate var shouldShowDefaultBrowserPromo: Bool { profile.prefs.intForKey(PrefsKeys.IntroSeen) == nil }
+    fileprivate var shouldShowDefaultBrowserPromo: Bool {
+        profile.prefs.intForKey(PrefsKeys.IntroSeen) == nil &&
+        DefaultBrowserExperiment.minPromoSearches() <= User.shared.searchCount
+    }
     fileprivate var shouldShowWhatsNewPageScreen: Bool { whatsNewDataProvider.shouldShowWhatsNewPage }
-    
+    fileprivate var shouldShowAPNConsentScreen: Bool {
+        EngagementServiceExperiment.isEnabled &&
+        EngagementServiceExperiment.minSearches() <= User.shared.searchCount &&
```

Comment:
Just realising, wouldn't it make sense to add this searches condition inside the `isEnabled`? Is there any case where they would be used separately?

This actually even also applies to the `DefaultBrowserExperiment`.

Reply:
I believe it's a matter of perspective. I agree with you with regards to the acknowledgment of components at the upper level to only know whether `isEnabled`, regardless of what (at that moment) the meaning is.
On the other hand, it may be easier to have a clearer condition straight at the UI level without having to dig deep.
At the time of writing the code, I purely took inspiration from the `DefaultBrowserExperiment` as that one came first and wanted the code to be as consistent as possible.

Reply:
Got it, indeed that is already what `DefaultBrowserExperiment` does, it just seems unnecessary to me if they are always used together and making it more concise would be more readable, even if you would need to jump inside to understand what it's doing, this would already most likely be done now if debugging.

Also not a blocker since it was a logic already in place.

Reply:
Would leave it as is, for now, we may think of refactoring and moving it all together as a separate task perhaps.